### PR TITLE
[DiffDriveController] Use generate parameter library

### PR DESCRIPTION
**Pull request by pac48**
This is an example PR of converting the parameter usage in differential drive controller to use https://github.com/PickNikRobotics/generate_parameter_library

### CODE REVIEW

File: diff_drive_controller/CMakeLists.txt [hunk 1]
```diff
@@ -30,6 +30,12 @@ foreach(Dependency IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
   find_package(${Dependency} REQUIRED)
 endforeach()
 
+find_package(generate_parameter_library REQUIRED)
+
+generate_parameter_library(diff_drive_controller_parameters
+  src/parameters.yaml
```
Comment (tylerjw) [inline diff_drive_controller/CMakeLists.txt:36]: Please rename this to `force_torque_sensor_broadcaster_parameters.yaml`.

Having many files in the repo with the name `parameters.yaml` will make searching for the correct file annoying.

Comment (pac48) [reply diff_drive_controller/CMakeLists.txt:36]: I think you meant `diff_drive_controller_parameter.yaml`. I will make the chnage

File: ros2_controllers-not-released.rolling.repos [hunk 2]
```diff
@@ -1 +1,5 @@
 repositories:
+  generate_parameter_library:
+    type: git
+    url: https://github.com/picknikrobotics/generate_parameter_library.git
+    version: main
```
Comment (destogl) [inline ros2_controllers-not-released.rolling.repos:5]:
```suggestion
```

File: ros2_controllers.rolling.repos [hunk 3]
```diff
@@ -15,3 +15,7 @@ repositories:
     type: git
     url: https://github.com/ros/angles.git
     version: ros2
+  generate_parameter_library:
+    type: git
+    url: https://github.com/picknikrobotics/generate_parameter_library.git
+    version: main
```
Comment (destogl) [inline ros2_controllers.rolling.repos:21]:
```suggestion
```

**Review by destogl — APPROVED** [review]

**Review by bmagyar — APPROVED** [review]
finally! :D

**Pull request merged**